python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -143,18 +143,62 @@ class DataFrame(object):
     3 3 0.3
     """
 
-    def __init__(self, name_series=None, index=None):
+    def __init__(self, data=None, index=None, columns=None):
+        keys = index
         if index is None:
             index = RangeIndex(start=0)
         self._index = as_index(index)
         self._size = len(index)
         self._cols = OrderedDict()
         # has initializer?
-        if name_series is not None:
-            if isinstance(name_series, dict):
-                name_series = name_series.items()
-            for k, series in name_series:
-                self.add_column(k, series, forceindex=index is not None)
+        added_columns = set()
+
+        if data is not None:
+            if isinstance(data, dict):
+                data = data.items()
+            elif utils.is_list_like(data) and len(data) > 0:
```

Comment:
Spoke offline with @galipremsagar regarding this chunk of code. Got clarification that in this case, `data` is a list of rows (not columns). Suggested making this chunk of code a private method that is called by `__init__()` since the constructor is getting large-ish

Reply:
@shwina made the refactor + added changes to `values` property